test/functional/wallet_timelock.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from test_framework.address import ADDRESS_BCRT1_UNSPENDABLE
+from test_framework.messages import CTxOut
+from test_framework.messages import tx_from_hex
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+
+class WalletLocktimeTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def run_test(self):
+        node = self.nodes[0]
+
+        mtp_tip = node.getblockheader(node.getbestblockhash())["mediantime"]
+
+        self.log.info("Get new address with label")
+        label = "timelock⌛🔓"
+        address = node.getnewaddress(label=label)
+        script_pub_key = node.validateaddress(address)["scriptPubKey"]
+
+        self.log.info("Send to new address with locktime")
+        tx = node.sendtoaddress(ADDRESS_BCRT1_UNSPENDABLE, 5)
+        tx = tx_from_hex(node.getrawtransaction(tx))
+        tx.vout = [CTxOut(5000, bytes.fromhex(script_pub_key))]
+        tx.nLockTime = mtp_tip - 1
+        tx_hex = node.signrawtransactionwithwallet(tx.serialize().hex())["hex"]
+        self.generateblock(node, ADDRESS_BCRT1_UNSPENDABLE, [tx_hex])
```

Comment:
In fa3e73af050e21b3fbcd7a269f4255abbf6183e3 "wallet: Avoid dropping confirmed coins"

This is way more complicated than it needs to be. We have RPCs that allow the locktime to be set. You can use `send` with `locktime` in the options.

Reply:
Good point, might adjust in the next force push or in a follow-up

Reply:
Thanks, fixed.